The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8254654783/job/22579303675) [(plain)](https://github.com/rust-lang/rust/commit/21ab40096b688f1a307964c3bf1ee7321d7b8465/checks/22579303675/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=ChrisDenton
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_5a4bc181-0f08-41e1-aab3-2166bf5e4cb5
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=1a55ab6ee7371271d48d6411f362170d076e4746
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_5a4bc181-0f08-41e1-aab3-2166bf5e4cb5
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_5a4bc181-0f08-41e1-aab3-2166bf5e4cb5
GITHUB_TRIGGERING_ACTOR=ChrisDenton
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/122401/merge
GITHUB_WORKFLOW_SHA=1a55ab6ee7371271d48d6411f362170d076e4746
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_20_X64=/opt/hostedtoolcache/go/1.20.14/x64
---
ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1

ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
           python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
           # check library crates on all tier 1 targets
           python3 ../x.py check --stage 0 core alloc std --stage 0 --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc && \
           python3 ../x.py build --stage 0 src/tools/build-manifest && \
           python3 ../x.py test --stage 0 src/tools/compiletest && \
           python3 ../x.py test --stage 0 core alloc std test proc_macro && \
           # Build both public and internal documentation.
---
#13 3.944 Building wheels for collected packages: reuse
#13 3.945   Building wheel for reuse (pyproject.toml): started
#13 4.335   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.336   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.336   Stored in directory: /tmp/pip-ephem-wheel-cache-chhkv125/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.341 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.365   Attempting uninstall: setuptools
#13 4.370     Found existing installation: setuptools 59.6.0
#13 4.370     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
DirectMap4k:      274368 kB
DirectMap2M:     8114176 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors &&            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu &&            python3 ../x.py check --stage 0 core alloc std --stage 0 --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc &&            python3 ../x.py clippy compiler -Aclippy::all -Dclippy::correctness &&            python3 ../x.py build --stage 0 src/tools/build-manifest &&            python3 ../x.py test --stage 0 src/tools/compiletest &&            python3 ../x.py test --stage 0 core alloc std test proc_macro &&            RUSTDOCFLAGS="--document-private-items --document-hidden-items" python3 ../x.py doc --stage 0 library &&            mkdir -p /checkout/obj/staging/doc &&            cp -r build/x86_64-unknown-linux-gnu/doc /checkout/obj/staging &&            RUSTDOCFLAGS="--document-private-items --document-hidden-items" python3 ../x.py doc --stage 0 compiler &&            RUSTDOCFLAGS="--document-private-items --document-hidden-items" python3 ../x.py doc --stage 0 library/test &&            /scripts/validate-toolstate.sh &&            /scripts/validate-error-codes.sh &&            reuse --include-submodules lint &&            es-check es6 ../src/librustdoc/html/static/js/*.js &&            eslint -c ../src/librustdoc/html/static/.eslintrc.js ../src/librustdoc/html/static/js/*.js &&            eslint -c ../src/tools/rustdoc-js/.eslintrc.js ../src/tools/rustdoc-js/tester.js &&            eslint -c ../src/tools/rustdoc-gui/.eslintrc.js ../src/tools/rustdoc-gui/tester.js
##[group]Building bootstrap
    Finished dev [unoptimized] target(s) in 0.05s
##[endgroup]
Ensuring the YAML anchors in the GitHub Actions config were expanded
---
+ python3 ../x.py check --stage 0 core alloc std --stage 0 --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc
##[group]Building bootstrap
    Finished dev [unoptimized] target(s) in 0.03s
##[endgroup]
error: the argument '--stage <N>' cannot be used multiple times

Usage: x.py check [OPTIONS] [PATHS]... [-- <ARGS>...]

For more information, try '--help'.
  local time: Tue Mar 12 19:34:38 UTC 2024
  network time: Tue, 12 Mar 2024 19:34:38 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/ci/docker/host-x86_64/mingw-check/Dockerfile
+++ b/src/ci/docker/host-x86_64/mingw-check/Dockerfile
@@ -41,7 +41,9 @@ COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
 ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 
 ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
-           # check library crates on all tier 1 targets
+           # Check library crates on all tier 1 targets.
+           # We disable optimized compiler built-ins because that requires a C toolchain for the target.
+           # We also skip the x86_64-unknown-linux-gnu target as it is well-tested by other jobs.
            python3 ../x.py check --stage 0 --set build.optimized-compiler-builtins=false core alloc std --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc && \
            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
            python3 ../x.py clippy compiler -Aclippy::all -Dclippy::correctness && \

--- a/src/ci/docker/host-x86_64/mingw-check/Dockerfile
+++ b/src/ci/docker/host-x86_64/mingw-check/Dockerfile
@@ -41,6 +41,8 @@ COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/
 ENV RUN_CHECK_WITH_PARALLEL_QUERIES 1
 
 ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
+           # check library crates on all tier 1 targets
+           python3 ../x.py check --stage 0 --set build.optimized-compiler-builtins=false core alloc std --target=aarch64-unknown-linux-gnu,i686-pc-windows-msvc,i686-unknown-linux-gnu,x86_64-apple-darwin,x86_64-pc-windows-gnu,x86_64-pc-windows-msvc && \
            python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu && \
            python3 ../x.py clippy compiler -Aclippy::all -Dclippy::correctness && \
            python3 ../x.py build --stage 0 src/tools/build-manifest && \

